packages/nuxt/src/app/plugins/http-client-hints/detect-browser.ts

```diff
@@ -0,0 +1,328 @@
+  ['silk', /\bSilk\/([0-9._-]+)\b/],
+  ['miui', /MiuiBrowser\/([0-9.]+)$/],
+  ['beaker', /BeakerBrowser\/([0-9.]+)/],
+  ['edge-chromium', /EdgA?\/([0-9.]+)/],
+  [
+    'chromium-webview',
+    /(?!Chrom.*OPR)wv\).*Chrom(?:e|ium)\/([0-9.]+)(:?\s|$)/
+  ],
+  ['chrome', /(?!Chrom.*OPR)Chrom(?:e|ium)\/([0-9.]+)(:?\s|$)/],
+  ['phantomjs', /PhantomJS\/([0-9.]+)(:?\s|$)/],
+  ['crios', /CriOS\/([0-9.]+)(:?\s|$)/],
+  ['firefox', /Firefox\/([0-9.]+)(?:\s|$)/],
+  ['fxios', /FxiOS\/([0-9.]+)/],
+  ['opera-mini', /Opera Mini.*Version\/([0-9.]+)/],
+  ['opera', /Opera\/([0-9.]+)(?:\s|$)/],
+  ['opera', /OPR\/([0-9.]+)(:?\s|$)/],
+  ['pie', /^Microsoft Pocket Internet Explorer\/(\d+\.\d+)$/],
+  ['pie', /^Mozilla\/\d\.\d+\s\(compatible;\s(?:MSP?IE|MSInternet Explorer) (\d+\.\d+);.*Windows CE.*\)$/],
+  ['netfront', /^Mozilla\/\d\.\d+.*NetFront\/(\d.\d)/],
+  ['ie', /Trident\/7\.0.*rv:([0-9.]+).*\).*Gecko$/],
+  ['ie', /MSIE\s([0-9.]+);.*Trident\/[4-7].0/],
+  ['ie', /MSIE\s(7\.0)/],
+  ['bb10', /BB10;\sTouch.*Version\/([0-9.]+)/],
+  ['android', /Android\s([0-9.]+)/],
+  ['ios', /Version\/([0-9._]+).*Mobile.*Safari.*/],
+  ['safari', /Version\/([0-9._]+).*Safari/],
+  ['facebook', /FB[AS]V\/([0-9.]+)/],
+  ['instagram', /Instagram\s([0-9.]+)/],
+  ['ios-webview', /AppleWebKit\/([0-9.]+).*Mobile/],
+  ['ios-webview', /AppleWebKit\/([0-9.]+).*Gecko\)$/],
+  ['curl', /^curl\/([0-9.]+)$/],
+  ['searchbot', SEARCHBOX_UA_REGEX]
+]
+const operatingSystemRules: OperatingSystemRule[] = [
+  ['iOS', /iP(hone|od|ad)/],
+  ['Android OS', /Android/],
+  ['BlackBerry OS', /BlackBerry|BB10/],
+  ['Windows Mobile', /IEMobile/],
+  ['Amazon OS', /Kindle/],
+  ['Windows 3.11', /Win16/],
+  ['Windows 95', /(Windows 95)|(Win95)|(Windows_95)/],
+  ['Windows 98', /(Windows 98)|(Win98)/],
+  ['Windows 2000', /(Windows NT 5.0)|(Windows 2000)/],
+  ['Windows XP', /(Windows NT 5.1)|(Windows XP)/],
+  ['Windows Server 2003', /(Windows NT 5.2)/],
+  ['Windows Vista', /(Windows NT 6.0)/],
+  ['Windows 7', /(Windows NT 6.1)/],
+  ['Windows 8', /(Windows NT 6.2)/],
+  ['Windows 8.1', /(Windows NT 6.3)/],
+  ['Windows 10', /(Windows NT 10.0)/],
+  ['Windows ME', /Windows ME/],
+  ['Windows CE', /Windows CE|WinCE|Microsoft Pocket Internet Explorer/],
+  ['Open BSD', /OpenBSD/],
+  ['Sun OS', /SunOS/],
+  ['Chrome OS', /CrOS/],
+  ['Linux', /(Linux)|(X11)/],
+  ['Mac OS', /(Mac_PowerPC)|(Macintosh)/],
+  ['QNX', /QNX/],
+  ['BeOS', /BeOS/],
+  ['OS/2', /OS\/2/]
+]
+
+export function detect (
+  userAgent?: string
+):
+  | BrowserInfo
+  | SearchBotDeviceInfo
+  | BotInfo
+  | NodeInfo
+  | ReactNativeInfo
+  | null {
+  if (userAgent) { return parseUserAgent(userAgent) }
+
+  if (
+    typeof document === 'undefined' &&
+        typeof navigator !== 'undefined' &&
+        navigator.product === 'ReactNative'
+  ) { return new ReactNativeInfo() }
+
+  if (typeof navigator !== 'undefined') { return parseUserAgent(navigator.userAgent) }
+
+  return getNodeVersion()
```

Comment:
Uhmm, this is only for server

Reply:
This library was just a copy/paste from another MIT one, check the license, I guess it would be better to use that library or just port and include it to unjs.

Reply:
> Uhmm, this is only for server

Yeah I mean on SSR on non node runtime, we'd want something different than `getNodeVersion` (ie return the server runtime version universally ?)